### PR TITLE
Replace Travis CI with GitHub Actions

### PR DESCRIPTION
**Pull request by tnir**
We do not see any new builds on Travis CI (.org) https://travis-ci.org/github/KnapsackPro/knapsack/builds (and the project is not found on .com https://travis-ci.com/github/KnapsackPro/knapsack).

Blocked by #114

### Changes

- The base OS image was upgraded from Ubuntu 16.04 LTS to 20.04 LTS.

### Announcement from Travis

https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
https://blog.travis-ci.com/oss-announcement

### Counterpart PRs

- https://github.com/KnapsackPro/docs.knapsackpro.com/pull/82

Signed-off-by: Takuya Noguchi <takninnovationresearch@gmail.com>

### CODE REVIEW

**Comment by tnir** [comment]
This PR is only for `push` or PR for `master`, https://github.com/tnir/knapsack/runs/2865274937 shows the results expected with this PR to you. Note that job with Ruby 2.1 fails.

Ruby 2.1 is preserved in this PR since it has been preserved even in #110 and/or on d4f2c80bc03a5ef86f0df9aa2da05b665d23193a.

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    - name: Run rake task for the first CI node
+      run: CI_NODE_TOTAL=2 CI_NODE_INDEX=0 KNAPSACK_TEST_FILE_PATTERN="test_examples/**{,/*/**}/*_test.rb" bundle exec rake knapsack:minitest
+    - name: Run rake task for the second CI node
+      run: CI_NODE_TOTAL=2 CI_NODE_INDEX=1 KNAPSACK_TEST_FILE_PATTERN="test_examples/**{,/*/**}/*_test.rb" bundle exec rake knapsack:minitest
+
+    - name: Check passing arguments to minitest. Run verbose tests
+      run: |
+        KNAPSACK_TEST_FILE_PATTERN="test_examples/**{,/*/**}/*_test.rb" bundle exec rake "knapsack:minitest[--verbose]"
+        if [ "$TRAVIS_RUBY_VERSION" == "1.9.3" ]; then KNAPSACK_TEST_FILE_PATTERN="test_examples/**{,/*/**}/*_test.rb" bin/knapsack minitest "--verbose"; fi
+        if [ "$TRAVIS_RUBY_VERSION" != "1.9.3" ]; then KNAPSACK_TEST_FILE_PATTERN="test_examples/**{,/*/**}/*_test.rb" bin/knapsack minitest "--verbose --pride"; fi
```
Comment (tnir) [inline .github/workflows/ruby.yml:79]: We can partially revert these lines (#77):
```suggestion
        KNAPSACK_TEST_FILE_PATTERN="test_examples/**{,/*/**}/*_test.rb" bin/knapsack minitest "--verbose --pride"
```
Or fully revert all of #77 in another PR.

Comment (tnir) [reply .github/workflows/ruby.yml:79]: > Or fully revert all of #77 in another PR.

Full reverting $77 is prepared in #113.

Comment (ArturT) [reply .github/workflows/ruby.yml:79]: I've merged your PR https://github.com/KnapsackPro/knapsack/pull/113

Could you apply proper changes here to remove conditional for `TRAVIS_RUBY_VERSION`?

**Comment by ArturT** [comment]
Could you fix the conflict in `.travis.yml`? It was changed in [your PR](https://github.com/KnapsackPro/knapsack/pull/113) that I've merged. You could simply remove this file again to resolve the conflict.

I'm wondering what to do with [failing build](https://github.com/tnir/knapsack/runs/2865274937) in Ruby 2.1.

On one hand [Ruby 2.1 support has ended](https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/) but on the other hand, I don't want to make the hard requirement of >= Ruby 2.2 to force users to use newer Ruby if they have legacy projects.

Regarding the error:

```
/opt/hostedtoolcache/Ruby/2.1.9/x64/lib/ruby/2.1.0/minitest/unit.rb:26:in `const_missing': uninitialized constant MiniTest::Test (NameError)
	from /home/runner/work/knapsack/knapsack/lib/knapsack/adapters/minitest_adapter.rb:24:in `bind_time_tracker'
```

Maybe we need a similar condition for Ruby 2.1 as [here for Ruby 1.9.3](https://github.com/KnapsackPro/knapsack/pull/77/files#diff-9063657c1a63bf14d4e50954446740bbb50540afe58f02bd0bf46cdd37f723e5R5).

[Ruby 2.1.10 seems to be the latest version available](https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/) for 2.1.x.

Maybe something like that?

```ruby
if RUBY_VERSION == "2.1.10"
  unless defined? MiniTest::Test
    MiniTest::Test = Minitest::Test
  end
end
```

**Comment by tnir** [comment]
> Could you fix the conflict in .travis.yml? It was changed in your PR that I've merged. You could simply remove this file again to resolve the conflict.

Yes, I did it.

> I'm wondering what to do with failing build in Ruby 2.1.

The tweak for minitest 4 (which had been bundled with Ruby 1.9 through 2.1) introduced in #26 does bad. We expected minitest 5 in #16, but we could not have been using minitest 5.x on CI (with Ruby 2.1 or earlier). We can merge #114 first.

**Comment by ArturT** [comment]
I've merged #114 and I merged this PR as well but CI build for [Ruby 2.1 still was failing](https://github.com/KnapsackPro/knapsack/runs/2883581100?check_suite_focus=true). 

I decided to drop support for [Ruby 2.1 here](https://github.com/KnapsackPro/knapsack/pull/115).

**Comment by ArturT** [comment]
I've released [knapsack 3.0.0](https://github.com/KnapsackPro/knapsack/blob/master/CHANGELOG.md#300).

Thanks @tnir for your help. :)

**Pull request merged**